esmvaltool/recipes/hydrology/recipe_pcrglobwb.yml

```diff
@@ -42,87 +31,44 @@ preprocessors:
     climate_statistics:
       operator: mean
       period: day
-  preproc_era5_pr: &preproc_era5_pr
+  preproc_pr: &preproc_pr
     <<: *crop_basin
-    <<: *era5_grid
     <<: *precipitation_unit
-  preproc_era5_tas: &preproc_era5_tas
+  preproc_tas: &preproc_tas
     <<: *crop_basin
-    <<: *era5_grid
     <<: *temperature_unit
-  preproc_era5_pr_clim:
-    <<: *preproc_era5_pr
+  preproc_pr_clim:
+    <<: *preproc_pr
     <<: *climatology
-  preproc_era5_tas_clim:
-    <<: *preproc_era5_tas
-    <<: *climatology
-  preproc_era_interim_pr: &preproc_era_interim_pr
-    <<: *crop_basin
-    <<: *era_interim_grid
-    <<: *precipitation_unit
-  preproc_era_interim_tas: &preproc_era_interim_tas
-    <<: *crop_basin
-    <<: *era_interim_grid
-    <<: *temperature_unit
-  preproc_era_interim_pr_clim:
-    <<: *preproc_era_interim_pr
-    <<: *climatology
-  preproc_era_interim_tas_clim:
-    <<: *preproc_era_interim_tas
+  preproc_tas_clim:
+    <<: *preproc_tas
     <<: *climatology
 
 diagnostics:
-  diagnostic_era_interim:
+  diagnostic_daily:
     description: PCR-GLOBWB input preprocessor for ERA-Interim data
     additional_datasets:
-      - {dataset: ERA-Interim, project: OBS6, tier: 3, type: reanaly, version: 1}
+      - {dataset: ERA-Interim, project: OBS6, tier: 3, type: reanaly, version: 1, mip: day}
+      - {dataset: ERA5, project: OBS6, tier: 3, type: reanaly, version: 1, mip: day}
     variables:
-      pr: &var_era_interim
-        mip: day
+      pr:
         start_year: 2002
         end_year: 2016
-        preprocessor: preproc_era_interim_pr
-      pr_climatology: &var_era_interim_clim
-        mip: day
-        start_year: 1990  # it should be possible to specify different years for the climatology
-        end_year: 2002
-        short_name: pr
-        preprocessor: preproc_era_interim_pr_clim
+        preprocessor: preproc_pr
       tas:
-        <<: *var_era_interim
-        preprocessor: preproc_era_interim_tas
-      tas_climatology:
-        <<: *var_era_interim_clim
-        short_name: tas
-        preprocessor: preproc_era_interim_tas_clim
-    scripts:
-      script:
-        script: hydrology/pcrglobwb.py
-        basin: Rhine
-
-  diagnostic_era5:
-    description: PCR-GLOBWB input preprocessor for ERA5 data
-    additional_datasets:
-      - {dataset: ERA5, project: OBS6, tier: 3, type: reanaly, version: 1}
-    variables:
-      pr: &var_era5
-        mip: day
         start_year: 2002
```

Comment:
To make it shorter, we can use the same year range in pr.

Reply:
I guess it's a matter of taste. This saves 2 line of code, but it makes the recipe more complicated to read.